scalding-core/src/main/scala/com/twitter/scalding/Config.scala

```diff
@@ -51,6 +51,20 @@ trait Config extends Serializable {
     }
 
   /**
+   * Add files to be localized to the config. Intended to be used by user code.
+   * @param cachedFiles CachedFiles to be added
+   * @return new Config with cached files
+   */
+  def addDistributedCacheFiles(cachedFiles: CachedFile*): Config =
```

Comment:
Hmm, I'm a little conflicted on this. My preference would be a method on `Execution` like `def addDistributedCacheFile(path: String): Execution[CachedFile]` or something similar. That way, the user doesn't have to worry about the `Mode`. Internally, we can just use the `Mode` in `Execution#run` to decide whether to mutate the Hadoop `Configuration`.